extensions/metrics/README.md

```diff
@@ -0,0 +1,202 @@
+# @loopback/extension-metrics
+
+This module contains a component to report metrics for
+[Prometheus](https://prometheus.io/).
+
+## Installation
+
+```sh
+npm install --save @loopback/extension-metrics
+```
+
+## Basic use
+
+The component should be loaded in the constructor of your custom Application
+class.
+
+Start by importing the component class:
+
+```ts
+import {MetricsComponent} from '@loopback/extension-metrics';
+```
+
+In the constructor, add the component to your application:
+
+```ts
+this.component(MetricsComponent);
+```
+
+By default, Metrics route is mounted at `/metrics`. This path can be customized
+via Metrics configuration as follows:
+
+```ts
+this.bind(MetricsBindings.CONFIG).to({
+  endpoint: {
+    basePath: '/metrics',
+  },
+  defaultMetrics: {
+    options: {
+      timeout: 5000,
+    },
+  },
+});
+```
+
+## Metrics Collected
+
+There are three types of metrics being collected by this component:
+
+1. Node.js metrics - built-in by https://github.com/siimon/prom-client
+2. LoopBack method invocations - built-in by this module using an interceptor
+3. Metrics by the application code or other modules
+
+## Pull vs Push
+
+Prometheus supports two modes to collect metrics:
+
+- pull - scaping from metrics http endpoint exposed by the system being
+  monitored
+- push - pushing metrics from the system being monitored to a push gateway
+
+See
+https://prometheus.io/docs/introduction/faq/#why-do-you-pull-rather-than-push
+
+## Try it out
+
+```sh
+npm run demo
+```
+
+### /metrics endpoint
+
+http://localhost:3000/metrics returns metrics in plain text format. It includes
+information for the Node.js process as well as LoopBack method invocations.
+
+```
+# HELP process_cpu_user_seconds_total Total user CPU time spent in seconds.
```

Comment:
This example is a long wall of text hiding the relevant content following the code, can you hide it using `<details>` tag please?

Syntax:

```markdown
<details>
<summary>Example of plain text data</summary>
<pre><code>
# HELP process_cpu_user_seconds_total Total user CPU time spent in seconds.
# TYPE process_cpu_user_seconds_total counter
process_cpu_user_seconds_total 0.132181 1564508354524
...
</code></pre>
</details>
```

Live demo:

<details>
<summary>Example of plain text data</summary>
<pre><code>
# HELP process_cpu_user_seconds_total Total user CPU time spent in seconds.
# TYPE process_cpu_user_seconds_total counter
process_cpu_user_seconds_total 0.132181 1564508354524
...
</code></pre>
</details>